modules/scripts/startup-script/examples/install_ansible.sh

```diff
@@ -30,11 +30,17 @@ apt_wait() {
 	fi
 }
 
-if ! command -v ansible-playbook >/dev/null 2>&1; then
+if [ ! -h /usr/bin/ansible-playbook ] || [ ! -f /usr/bin/ansible-playbook ]; then
 	if [ -f /etc/centos-release ] || [ -f /etc/redhat-release ] || [ -f /etc/oracle-release ] || [ -f /etc/system-release ]; then
-		yum -y install epel-release
-		yum -y install ansible
-
+		if [ ! -f /bin/pip ]; then
```

Comment:
The CentOS 7 copy of pip is so old it can't even successfully upgrade itself (it will aim too high for Python 3.6) and it also cannot install `google-cloud-*` libraries (it will aim too high for 3.6).

You have to specify which releases you want explicitly. That seems like a bad experience waiting to happen. Ideally I would test what the system version of pip is here rather than just the existence of pip. But I do, in fact, want the system pip to be recent as Step 0.

Reply:
See, e.g. #368 and its resolution